frontend/src/components/VAudioThumbnail/VAudioThumbnail.vue

```diff
@@ -60,7 +60,9 @@ export default defineComponent({
     },
   },
   setup(props) {
-    const { isHidden: shouldBlur } = useSensitiveMedia(props.audio)
+    const shouldBlur = computed(
+      () => useSensitiveMedia(props.audio).isHidden.value
+    )
```

Comment:
Calling `useSensitiveMedia` inside the computed callback could be risky if `useSensitiveMedia` provisions `ref` objects, which it does (`isHidden` itself is a `computed`).

Olga suggested the following fix in the issue:

> To fix this, we should pass the media as a ref to the use-sensitive-media composable.

I think this is the better fix and would require less code overall. To implement the fix, we'd just need to wrap `props` in `toRefs`:

```suggestion
    const { audio } = toRefs(props)
    const { isHidden: shouldBlur } = useSensitiveMedia(audio)
```

As Olga pointed out the issue, the problem with the current implementation is that `props.audio` is getting detached from updates to `props`, so `audio` is no longer reactive. The current implement is identical to destructuring from the raw `props` object, as opposed to using `props.audio` inside the ref. `toRefs` ensures that the new `audio` ref pulled from `toRefs(props)` is updated whenever the reactive `props` object is updated.

Reply:
I was under the impression that both solutions were functionally equivalent and had no idea about it being "risky". Could you elaborate on what the risk is and how? Is that related to the memory leak issue we had a while before?

Reply:
Yes! The memory leak issue we saw before is exactly it. Each time the computed re-runs, the call to `useSensitiveMedia` provisions the computed refs it uses each time instead of reusing the existing ones, and those don't get garbage collected. It might not be visible on the server side because of how the page renders, but this one would be evident client side.

Here's the PR where we fixed it before: https://github.com/WordPress/openverse-frontend/pull/1747

I can't find the explanation we had for why it happened.

If the other implementation is preferable for some reason, you could always profile it locally and see whether it has the same issue.